OracleGoldenGate/21c/install-deployment.sh

```diff
@@ -44,6 +44,26 @@ function ogg_installer_setup() {
     unzip -q "${INSTALLER}" -d "/tmp/installer" || abort "Unzip operation failed for '${INSTALLER}'"
     chmod -R o=g-w             "/tmp/installer"
 }
+##
+##
+## o g g _ i n s t a l l _ o p t i o n
+##
+function ogg_install_option() {
+  # Get the path to the fastcopy.xml file that contains the metadata for the install option.
+  local fast_copy_file
+  fast_copy_file=$(find /tmp/installer -name fastcopy.xml)
+
+  # Get the xml line that needs to be parsed.
+  local xml_line
+  xml_line=$(grep -h 'TOPLEVEL_COMPONENT NAME' "${fast_copy_file}")
+
+  # Match string example:   <TOPLEVEL_COMPONENT NAME="oracle.oggcore.services" INSTALL_TYPE="ora21c" PLATFORM="Linux">
+  local regex
+  regex='.*<TOPLEVEL_COMPONENT NAME=.* INSTALL_TYPE="(.*)" .*'
+
+  [[ $xml_line =~ $regex ]]                     || abort "Could not find INSTALL_TYPE in the file '${fast_copy_file}'"
+  INSTALL_OPTION="${BASH_REMATCH[1]}"
```

Comment:
Setting values for GLOBAL variables as a side effect of a function is a bad practice. Use echo and:

```
INSTALL_OPTION=$(ogg_install_option)
```

on line 80